### ISSUE TITLE
Hard resetting while in-game softlocks the game

### ISSUE THREAD

**Issue by Elomavi** [TerryCavanagh/VVVVVV#856]
In versions 2.3.4 and prior, hard resetting while in-game would send the player back out to the main menu. For the most part, everything functioned normally but this could lead to some buggy behavior if you performed the right setup. This lead to a change in v2.3.5 where 

> hardreset now resets ingame_titlemode.

However, if you attempt to hard reset while in-game in v2.3.5, you will be taken to a completely blank main menu. Getting put into this menu completely softlocks you with no way out except to restart your game. You can do a couple of things, but they all just end up in black screens and you won't be able to resume normal gameplay.

To replicate this bug:
1. Continue from any save file or start a new game so you're in-game.
2. Press Escape to access options
3. Press "Clear Main Game Data"

**Comment by InfoTeddy** [comment]
On `master`, I committed 2770353142c4205f14604499adc5ae4303ad83b6 but forgot to apply it to the 2.3 branch, so 2.3.5 still takes you back to the main menu. Which is unfortunate; I should have tested it. Whoops.

**Comment by InfoTeddy** [comment]
Also btw you were conflating the `hardreset` function and deleting save data, which are not the same. It's just that I added another check where the game asserts and doesn't create the main menu if you're `ingame_titlemode`.

Anyways I pushed the commit to the 2.3 branch (cc3dc8d32912ea0663fef0af12a5d93bb71d8e89), along with another backported bugfix that I forgot to push for 2.3.5. And yes, I tested both of them this time. Unfortunately Ethan is apparently going to be out until the 26th, so I don't think 2.3.6 is coming until then.
